WcaOnRails/app/models/competition.rb

```diff
@@ -742,23 +742,21 @@ def self.search(query, params: {})
       competitions = competitions.where("CAST(CONCAT(year,'-',endMonth,'-',endDay) as Date) <= ?", end_date)
     end
 
-    if query.present?
-      sql_query = "%#{query}%"
-      competitions = competitions.where("id LIKE :sql_query OR name LIKE :sql_query OR cellName LIKE :sql_query OR cityName LIKE :sql_query OR countryId LIKE :sql_query", sql_query: sql_query).order(year: :desc, month: :desc, day: :desc)
+    query&.split&.each do |part|
+      like_query = %w(id name cellName cityName countryId).map { |column| column + " LIKE :part" }.join(" OR ")
+      competitions = competitions.where(like_query, part: "%#{part}%")
     end
 
-    if params[:sort].present?
-      params[:sort].split(",").each do |field|
-        order = field.start_with?("-") ? :desc : :asc
+    (params[:sort] || "-start_date").split(",").each do |field|
```

Comment:
This is the only weird part. And it was weird before. It looks like there is/was a loop to add order clauses to the query. Why you'd ever have several of those on the same query I have no idea.

Anyway, it looks like you added `-start_date` as a default value to `params[:sort]`, which will make it sort descending by start date, rather than being unsorted. Which is probably an improvement.

Or did I misunderstand something?